cpp/include/cudf/strings/detail/merge.cuh

```diff
@@ -89,9 +89,8 @@ std::unique_ptr<column> merge(strings_column_view const& lhs,
   auto d_offsets = offsets_column->view().template data<int32_t>();
 
   // create the chars column
-  auto chars_column = strings::detail::create_chars_child_column(bytes, stream, mr);
-  // merge the strings
-  auto d_chars = chars_column->mutable_view().template data<char>();
+  rmm::device_uvector<char> chars(bytes, stream, mr);
```

Comment:
Why do we prefer a `device_uvector` instead of a raw `buffer` here, given that we want to hold character buffers at the end? (This pattern reoccurs.)

Reply:
On further thought, maybe it's just nicer to have the typed container than a raw buffer. It plays well with our sync/async vector factories and the `.release()` is fairly easy to deal with when you're ready for a buffer. I'm okay with keeping this, just interested in hearing your thoughts.